### DISCUSSION TITLE
fecth data from server/api, but not displayed in source code (SEO)

### DISCUSSION THREAD

**Discussion by JiatLn** [nuxt/nuxt#16268] [Questions]
- server\fn\index.ts
```ts
async function fetchCMS<T>(url: string, params: Record<string, string | number | undefined> = {}) {
  const { apiSecret } = useRuntimeConfig()
  return await $fetch<T>(url, {
    baseURL: `${apiSecret}/cms`,
    params,
    headers: {
      'Content-Type': 'application/json;charset=utf-8',
    },
    responseType: 'json',
  })
}
```

- server\api\article\list.ts
```ts
import type { RespRow } from '~~/types'
import type { ArticleItem } from '~~/types/cmd'
import { fetchCMS } from '~~/server/fn'

export default defineEventHandler(async (event) => {
  const query = getQuery(event) as any
  const response = await fetchCMS<RespRow<ArticleItem>>(`/article/articleList/${query.categoryId}`, {
    ...query,
  })
  return response
})
```

and use in client

- composables\cms.ts
```ts
export function getArticleList(categoryId: number, queryParams?: QueryParams) {
  return $fetch<RespRow<ArticleItem>>('/api/article/list', {
    params: {
      ...queryParams,
      categoryId,
    },
  })
}
```

the source code is follow:
```html
<!--v-if--></div><!--]--><!--]--><!--]--><!--]--></div></div></div><!--]--><!--]--></div>
```
![image](https://user-images.githubusercontent.com/16984236/203463775-fe221631-6026-4f8a-908b-dc94afeef02e.png)

Could you tell me how to fix it?

**Comment by mctweb** [comment]
You are using $fetch, which is client side. You can you useFetch, which is a wrapper for useAsyncData and $fetch. This can retrieve your data on the server.
https://nuxt.com/docs/api/composables/use-fetch

**Reply by JiatLn** [reply]
could i use `useFetch` in `/composables` not in setup script

**Reply by mctweb** [reply]
Yes if you want. Nuxt composables are available in other composables.

**Comment by TheAlexLichter** [comment]
The most important part of the code is missing 🙈
The question is *how* you use `getArticleList`.

I'd suggest you to call it inside `useAsyncData` and await the composable. Then, it should work fine.
